src/ccl_cls.c

```diff
@@ -887,27 +887,39 @@ static void get_k_interval(ccl_cosmology *cosmo,CCL_ClWorkspace *w,
 //clt1 -> tracer #1
 //clt2 -> tracer #2
 static double ccl_angular_cl_native(ccl_cosmology *cosmo,CCL_ClWorkspace *cw,int il,
-				    CCL_ClTracer *clt1,CCL_ClTracer *clt2,int * status)
+				    CCL_ClTracer *clt1,CCL_ClTracer *clt2,
+				    ccl_p2d_t *psp,int * status)
 {
   int clastatus=0, gslstatus;
   IntClPar ipar;
   double result=0,eresult;
   double lkmin,lkmax;
+  ccl_p2d_t *psp_use;
   gsl_function F;
   gsl_integration_workspace *w=gsl_integration_workspace_alloc(ccl_gsl->N_ITERATION);
 
+  if(psp==NULL) {
+    if (!cosmo->computed_power) ccl_cosmology_compute_power(cosmo, status);
+    // Return if compilation failed
+    if (!cosmo->computed_power) return NAN;
```

Comment:
It is handled by the python status check. However, to make the process faster I've changed things a bit so that, the first time this happens, it goes straight back to python without trying to compute anything else.